python/ml4ir/applications/ranking/tests/data/configs/feature_config.yaml

```diff
@@ -45,7 +45,7 @@ features:
     node_name: text_match_score
     trainable: true
     dtype: float
-    log_at_inference: false
+    log_at_inference: true
```

Comment:
@ducouloa I mentioned this to @jakemannix as well. This is fine (and even good) for now. But we should also think about decoupling the end to end testing from reading the features from the predictions file.

There are two reasons:
- At training time, this data is being read from tfrecords and then is getting converted into numpy/pandas objects. There is a lot of type conversion happening. More importantly, the data here already has all the feature processing applied. This is a good thing, but might not be later on.
- Less important, but still relevant - if we want to use this for training time validation (which we definitely should), we will have to write an entire CSV file full of duplicate feature values (feature values that are already available in the TFRecord). If we are training 50 models (for hyper param optimization), this means that we will be writing back 50x the input data size onto deepsea (or at the least the docker image). We should avoid this if possible. Unfortunately, there is no easy solution to this until we can read TFRecord files directly in java. I don't recommend delving into that now.

@ducouloa Do NOT implement any of the changes recommended here for now. We need to think more about this. As I said, this is more than good for now.

Reply:
@lastmansleeping can we open a work item to track the issues around log_at_inference usage in tests/prod configs?

Reply:
Yep, let me do that.